### PR TITLE
SQL and Mac OSx Changes.

### PR DESCRIPTION
**Pull request by brahmastra2016**
Zeroblob SQL query unique constrain fix.
Mac OSx Support added.
     a) Diagnostic "--sysinfo" knob work both in GUI and CLI
     b) Cleans:
                    Rotated Logs.
                    Firefox.
                    Deep Scan (Delete DS_Store)
                   Recycle Bin

### CODE REVIEW

File: bleachbit/Special.py [hunk 1]
```diff
@@ -41,12 +41,12 @@ def __shred_sqlite_char_columns(table, cols=None, where=""):
     """Create an SQL command to shred character columns"""
     cmd = ""
     if cols and options.get('shred'):
+        #cmd += "update %s set %s %s;" % \
```
Comment (az0) [inline bleachbit/Special.py]: I would like to address the SQL error in another way, so would you please remove it from this pull request?

**Comment by brahmastra2016** [comment]
i had tried that approach 'if col !="url"' but then it would fail with the same  unique constrain error at 
"update moz_host = zeroblob(lenght(host))"

**Comment by az0** [comment]
I thought I suggested a different approach.

**Comment by brahmastra2016** [comment]
sorry i read my own code and got confused.

i change my pull request for now.

just a update .....
i saw that Firefox database (place.sqlite) changes and URL is no more set to UNIQUE 
now the problem is just host in moz_hosts

screen shot 2016-02-19 at 6 32 50 pm

**Comment by brahmastra2016** [comment]
Will Reopen new with other Mac OSx Changes

**Pull request closed without merging**